### PR TITLE
Remove a few pieces of unnecessary code in the tests of sparse block dot.

### PR DESCRIPTION
**Pull request by adbrebs**
@bouthilx

### CODE REVIEW

File: theano/sandbox/tests/test_blocksparse.py [hunk 1]
```diff
@@ -122,30 +114,6 @@ def gemv_numpy3(o, W, h, iIdx, oIdx):
         return o
 
     @staticmethod
-    def gemv_data2():
```
Comment (bouthilx) [inline theano/sandbox/tests/test_blocksparse.py]: @adbrebs  Do you remember what it was for?

File: theano/sandbox/tests/test_blocksparse.py [hunk 2]
```diff
@@ -41,14 +41,6 @@ def gemv_data():
         outputWindowSize = 3
         batchSize = 2
 
-#        nInputBlock = 2
```
Comment (bouthilx) [inline theano/sandbox/tests/test_blocksparse.py]: Oh my, thank you! ;)

File: theano/sandbox/tests/test_blocksparse.py (hunk 1 above)
Comment (adbrebs) [inline theano/sandbox/tests/test_blocksparse.py]: Somehow a more realistic data generation oO.

**Comment by bouthilx** [comment]
OK, I'll merge when travis is done. I'll ping you to rebase #3361.

**Comment by adbrebs** [comment]
@bouthilx Why did this PR force us to rebase #3361? I don't see any conflict.

**Comment by bouthilx** [comment]
@adbrebs You rename the file in the other PR.

**Pull request merged**